pkg/sensors/tracing/generickprobe.go

```diff
@@ -315,6 +325,17 @@ func preValidateKprobes(name string, kprobes []v1alpha1.KProbeSpec) error {
 			}
 		}
 
+		if strings.HasPrefix(f.Call, "list:") {
```

Comment:
could there be functions in the kernel that this prefix collides with?

Also, what is the meaning of the rest of the fields when the call has a `list:` prefix. 

Looking at the kprobe definition, besides `Call`, we have: `Return`,  `Syscall`, `Args`,  `ReturnArg`, `ReturnArgAction`, and `Selectors`. For `Return`, `Syscall` and `Selectors`, we can say that whatever is defined there applies to all calls.

Not so sure about `Args`, `ReturnArg`, and `ReturnArgAction`. I think we could leave it as is. The argument fields will be applied to all calls in the list, and we assume the user is knows what they are doing. In the future, we could also provide `list:` arguments to specify them separately.

So I _think_ it works.  We should definitively document this though. I guess we would want to do this as a followup, so I created an issue: https://github.com/cilium/tetragon/issues/1365.

Reply:
> could there be functions in the kernel that this prefix collides with?

there's the ':' and that can't be part of kernel symbol

might be problem for C++ symbols for uprobes I guess.. but we could force single ':' compared to C++ '::'

> 
> Also, what is the meaning of the rest of the fields when the call has a `list:` prefix.
> 
> Looking at the kprobe definition, besides `Call`, we have: `Return`, `Syscall`, `Args`, `ReturnArg`, `ReturnArgAction`, and `Selectors`. For `Return`, `Syscall` and `Selectors`, we can say that whatever is defined there applies to all calls.
> 
> Not so sure about `Args`, `ReturnArg`, and `ReturnArgAction`. I think we could leave it as is. The argument fields will be applied to all calls in the list, and we assume the user is knows what they are doing. In the future, we could also provide `list:` arguments to specify them separately.
> 

the list in 'call' field is just sort of values iterator and the rest of the fields from kprobe applies to each value,
as if it was separate kprobe definition with 'call: valueX' from the list

> So I _think_ it works. We should definitively document this though. I guess we would want to do this as a followup, so I created an issue: #1365.

yes, I wanted to get some agrement first before I'd touch docs ;-)

Reply:
> might be problem for C++ symbols for uprobes I guess.. but we could force single ':' compared to C++ '::'

We are only applying it to kprobes, so I think that's fine.

> the list in 'call' field is just sort of values iterator and the rest of the fields from kprobe applies to each value,
as if it was separate kprobe definition with 'call: valueX' from the list

👍🏼